[libtasn1] Build error

<!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
<!-- ⁉ You can find its location at the end of the console output. ⁉ -->

**Host Environment**

- OS: Debian bookworm/sid
- Compiler: android-ndk-r25c
- Triplet: arm64-android
- vcpkg version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75

**To Reproduce**

Steps to reproduce the behavior:

`./vcpkg install libtasn1:arm64-android`

**Failure logs**

- console output:

```
❯ ./vcpkg install libtasn1:arm64-android
Computing installation plan...
The following packages will be built and installed:
    libtasn1[core]:arm64-android -> 4.19.0
Detecting compiler hash for triplet arm64-android...
Restored 0 package(s) from /home/arlen/.cache/vcpkg/archives in 4.31 us. Use --debug to see more details.
Installing 1/1 libtasn1:arm64-android...
Building libtasn1[core]:arm64-android...
warning: -- Using community triplet arm64-android. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /home/arlen/vcpkg/triplets/community/arm64-android.cmake
-- Using cached libtasn1-4.19.0.tar.gz.
-- Extracting source /home/arlen/vcpkg/downloads/libtasn1-4.19.0.tar.gz
-- Applying patch msvc_fixes.patch
-- Using source at /home/arlen/vcpkg/buildtrees/libtasn1/src/v4.19.0-312d3e2af8.clean
-- Found external ninja('1.11.1').
-- Getting CMake variables for arm64-android-dbg
-- Getting CMake variables for arm64-android-rel
-- Generating configure for arm64-android
-- Finished generating configure for arm64-android
-- Configuring arm64-android-dbg
-- Configuring arm64-android-rel
-- Building arm64-android-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/make V=1 -j 17 -f Makefile all
    Working Directory: /home/arlen/vcpkg/buildtrees/libtasn1/arm64-android-dbg/
    See logs for more information:
      /home/arlen/vcpkg/buildtrees/libtasn1/build-arm64-android-dbg-out.log
      /home/arlen/vcpkg/buildtrees/libtasn1/build-arm64-android-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:149 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  ports/libtasn1/portfile.cmake:46 (vcpkg_install_make)
  scripts/ports.cmake:147 (include)


error: building libtasn1:arm64-android failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libtasn1
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[libtasn1]+Build+error
Include '[libtasn1] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2023-01-24-8a88d63f241d391772fbde69af9cab96c3c64c75
    vcpkg-scripts version: 69a9c8561 2023-02-21 (54 minutes ago)
```
- log file:
```
In file included from ../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/strverscmp.c:26:
In file included from ./string.h:41:
In file included from /home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/string.h:187:
/home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/bits/fortify/string.h:95:47: error: use of undeclared identifier '__USE_FORTIFY_LEVEL'
        __clang_error_if(__bos_unevaluated_le(__bos(dst), __builtin_strlen(src)),
                                              ^
../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/cdefs.h:144:48: note: expanded from macro '__bos'
#define __bos(ptr) __builtin_object_size (ptr, __USE_FORTIFY_LEVEL > 1)
                                               ^
In file included from ../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/strverscmp.c:26:
In file included from ./string.h:41:
In file included from /home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/string.h:187:
/home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/bits/fortify/string.h:95:47: error: use of undeclared identifier '__USE_FORTIFY_LEVEL'
../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/cdefs.h:144:48: note: expanded from macro '__bos'
#define __bos(ptr) __builtin_object_size (ptr, __USE_FORTIFY_LEVEL > 1)
```
```
make[5]: Entering directory '/home/arlen/vcpkg/buildtrees/libtasn1/arm64-android-dbg/lib/gl'
/bin/bash ../../libtool  --tag=CC   --mode=compile /home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -DHAVE_CONFIG_H -I. -I../.././../src/v4.19.0-312d3e2af8.clean/lib/gl -I../..   --target=aarch64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 -fvisibility=hidden -Wno-cast-qual -Wno-conversion -Wno-float-equal -Wno-sign-compare -Wno-undef -Wno-unused-function -Wno-unused-parameter -Wno-float-conversion -Wimplicit-fallthrough -Wno-pedantic -Wno-sign-conversion -Wno-type-limits -g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -g -O2 -fno-limit-debug-info -MT libgnu_la-c-ctype.lo -MD -MP -MF .deps/libgnu_la-c-ctype.Tpo -c -o libgnu_la-c-ctype.lo `test -f 'c-ctype.c' || echo '../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/'`c-ctype.c
/bin/bash ../../libtool  --tag=CC   --mode=compile /home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -DHAVE_CONFIG_H -I. -I../.././../src/v4.19.0-312d3e2af8.clean/lib/gl -I../..   --target=aarch64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 -fvisibility=hidden -Wno-cast-qual -Wno-conversion -Wno-float-equal -Wno-sign-compare -Wno-undef -Wno-unused-function -Wno-unused-parameter -Wno-float-conversion -Wimplicit-fallthrough -Wno-pedantic -Wno-sign-conversion -Wno-type-limits -g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -g -O2 -fno-limit-debug-info -MT libgnu_la-strverscmp.lo -MD -MP -MF .deps/libgnu_la-strverscmp.Tpo -c -o libgnu_la-strverscmp.lo `test -f 'strverscmp.c' || echo '../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/'`strverscmp.c
/bin/bash ../../libtool  --tag=CC   --mode=compile /home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -DHAVE_CONFIG_H -I. -I../.././../src/v4.19.0-312d3e2af8.clean/lib/gl -I../..   --target=aarch64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 -fvisibility=hidden -Wno-cast-qual -Wno-conversion -Wno-float-equal -Wno-sign-compare -Wno-undef -Wno-unused-function -Wno-unused-parameter -Wno-float-conversion -Wimplicit-fallthrough -Wno-pedantic -Wno-sign-conversion -Wno-type-limits -g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -g -O2 -fno-limit-debug-info -MT libgnu_la-unistd.lo -MD -MP -MF .deps/libgnu_la-unistd.Tpo -c -o libgnu_la-unistd.lo `test -f 'unistd.c' || echo '../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/'`unistd.c
libtool: compile:  /home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -DHAVE_CONFIG_H -I. -I../.././../src/v4.19.0-312d3e2af8.clean/lib/gl -I../.. --target=aarch64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 -fvisibility=hidden -Wno-cast-qual -Wno-conversion -Wno-float-equal -Wno-sign-compare -Wno-undef -Wno-unused-function -Wno-unused-parameter -Wno-float-conversion -Wimplicit-fallthrough -Wno-pedantic -Wno-sign-conversion -Wno-type-limits -g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -g -O2 -fno-limit-debug-info -MT libgnu_la-c-ctype.lo -MD -MP -MF .deps/libgnu_la-c-ctype.Tpo -c ../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/c-ctype.c -o libgnu_la-c-ctype.o
libtool: compile:  /home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -DHAVE_CONFIG_H -I. -I../.././../src/v4.19.0-312d3e2af8.clean/lib/gl -I../.. --target=aarch64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 -fvisibility=hidden -Wno-cast-qual -Wno-conversion -Wno-float-equal -Wno-sign-compare -Wno-undef -Wno-unused-function -Wno-unused-parameter -Wno-float-conversion -Wimplicit-fallthrough -Wno-pedantic -Wno-sign-conversion -Wno-type-limits -g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -g -O2 -fno-limit-debug-info -MT libgnu_la-strverscmp.lo -MD -MP -MF .deps/libgnu_la-strverscmp.Tpo -c ../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/strverscmp.c -o libgnu_la-strverscmp.o
libtool: compile:  /home/arlen/android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -DHAVE_CONFIG_H -I. -I../.././../src/v4.19.0-312d3e2af8.clean/lib/gl -I../.. --target=aarch64-none-linux-android21 -DANDROID -D_FORTIFY_SOURCE=2 -fvisibility=hidden -Wno-cast-qual -Wno-conversion -Wno-float-equal -Wno-sign-compare -Wno-undef -Wno-unused-function -Wno-unused-parameter -Wno-float-conversion -Wimplicit-fallthrough -Wno-pedantic -Wno-sign-conversion -Wno-type-limits -g -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -Wformat -Werror=format-security -fPIC -g -O2 -fno-limit-debug-info -MT libgnu_la-unistd.lo -MD -MP -MF .deps/libgnu_la-unistd.Tpo -c ../.././../src/v4.19.0-312d3e2af8.clean/lib/gl/unistd.c -o libgnu_la-unistd.o
mv -f .deps/libgnu_la-unistd.Tpo .deps/libgnu_la-unistd.Plo
mv -f .deps/libgnu_la-c-ctype.Tpo .deps/libgnu_la-c-ctype.Plo
make[5]: Leaving directory '/home/arlen/vcpkg/buildtrees/libtasn1/arm64-android-dbg/lib/gl'
```

<!-- ‼ **The console output by itself is not enough to investigate a build failure.** ‼ -->

**Additional context**

The issue looks similar to #27304 #27177, I have tried `git revert 60993b8ec5d5cec012d7b3dc78d39d892a997a62 --no-edit` and libiconv-1.16 built for arm64-android successfully. But revert the last change of port/libtasn1 doesn't work.

--- a/ports/libtasn1/clang-fortify.patch
+++ b/ports/libtasn1/clang-fortify.patch
@@ -1,0 +1,64 @@
+diff --git a/lib/gl/cdefs.h b/lib/gl/cdefs.h
+index 7b8ed5b..63574f9 100644
+--- a/lib/gl/cdefs.h
++++ b/lib/gl/cdefs.h
+@@ -140,6 +140,7 @@
+ #endif
+ 
+ 
++#ifndef __GNULIB_CDEFS
+ /* Fortify support.  */
+ #define __bos(ptr) __builtin_object_size (ptr, __USE_FORTIFY_LEVEL > 1)
+ #define __bos0(ptr) __builtin_object_size (ptr, 0)
+@@ -201,6 +202,8 @@
+       ? __ ## f ## _chk_warn (__VA_ARGS__, (__osz) / (__s))		      \
+       : __ ## f ## _chk (__VA_ARGS__, (__osz) / (__s))))		      \
+ 
++#endif
++
+ #if __GNUC_PREREQ (4,3)
+ # define __warnattr(msg) __attribute__((__warning__ (msg)))
+ # define __errordecl(name, msg) \
+diff --git a/lib/gl/libc-config.h b/lib/gl/libc-config.h
+index a56665b..876e1a1 100644
+--- a/lib/gl/libc-config.h
++++ b/lib/gl/libc-config.h
+@@ -137,8 +137,10 @@
+ # undef __attribute_returns_twice__
+ # undef __attribute_used__
+ # undef __attribute_warn_unused_result__
++# ifndef __GNULIB_CDEFS
+ # undef __bos
+ # undef __bos0
++# endif
+ # undef __errordecl
+ # undef __extension__
+ # undef __extern_always_inline
+@@ -147,21 +149,27 @@
+ # undef __fortified_attr_access
+ # undef __fortify_function
+ # undef __glibc_c99_flexarr_available
++# ifndef __GNULIB_CDEFS
+ # undef __glibc_fortify
+ # undef __glibc_fortify_n
++# endif
+ # undef __glibc_has_attribute
+ # undef __glibc_has_builtin
+ # undef __glibc_has_extension
+ # undef __glibc_likely
+ # undef __glibc_macro_warning
+ # undef __glibc_macro_warning1
++# ifndef __GNULIB_CDEFS
+ # undef __glibc_objsize
+ # undef __glibc_objsize0
+ # undef __glibc_safe_len_cond
+ # undef __glibc_safe_or_unknown_len
++# endif
+ # undef __glibc_unlikely
++# ifndef __GNULIB_CDEFS
+ # undef __glibc_unsafe_len
+ # undef __glibc_unsigned_or_positive
++# endif
+ # undef __inline
+ # undef __ptr_t
+ # undef __restrict

--- a/ports/libtasn1/portfile.cmake
+++ b/ports/libtasn1/portfile.cmake
@@ -1,7 +1,6 @@
-vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
-
 vcpkg_download_distfile(ARCHIVE
     URLS "https://ftp.gnu.org/gnu/libtasn1/libtasn1-${VERSION}.tar.gz"
+         "https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/libtasn1/libtasn1-${VERSION}.tar.gz"
     FILENAME "libtasn1-${VERSION}.tar.gz"
     SHA512 287f5eddfb5e21762d9f14d11997e56b953b980b2b03a97ed4cd6d37909bda1ed7d2cdff9da5d270a21d863ab7e54be6b85c05f1075ac5d8f0198997cf335ef4
 )
@@ -11,6 +10,7 @@ vcpkg_extract_source_archive(SOURCE_PATH
     SOURCE_BASE "v${VERSION}"
     PATCHES
         msvc_fixes.patch
+        clang-fortify.patch # ported from https://git.savannah.gnu.org/cgit/gnulib.git/commit/?id=522aea1093a598246346b3e1c426505c344fe19a
 )
 
 vcpkg_find_acquire_program(BISON)
@@ -31,7 +31,7 @@ if (VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     list(APPEND EXTRA_OPTS "CFLAGS=\$CFLAGS -DASN1_STATIC")
 endif()
 
-set(ENV{GTKDOCIZE} true)
+set(ENV{GTKDOCIZE} true) # true, the program
 vcpkg_configure_make(
     SOURCE_PATH "${SOURCE_PATH}"
     AUTOCONFIG

--- a/ports/libtasn1/vcpkg.json
+++ b/ports/libtasn1/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "libtasn1",
   "version": "4.19.0",
+  "port-version": 1,
   "description": "A secure communications library implementing the SSL, TLS and DTLS protocols",
   "homepage": "https://www.gnutls.org/",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4462,7 +4462,7 @@
     },
     "libtasn1": {
       "baseline": "4.19.0",
-      "port-version": 0
+      "port-version": 1
     },
     "libtcod": {
       "baseline": "1.23.1",

--- a/versions/l-/libtasn1.json
+++ b/versions/l-/libtasn1.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d8d108b8c34710e58f52569772f13a406893c6ee",
+      "version": "4.19.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "a1922bb6cb0484a7aff7e4907285fc7f6d98fe4e",
       "version": "4.19.0",
       "port-version": 0

